interview-questions.md

```diff
@@ -0,0 +1,55 @@
+
+Researched answer: JSX is a syntax extension to JavaScript and it produces React elements, enabling them to be rendered by the DOM. 
+
+2. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
+
+Your answer: Yarn, stands for, yet another resource negotiator. It is a resource negotiator created by facebook to add and track dependencies required for a particular web application to run.
+
+Researched answer: Yarn is a package manager that also serves as a project manager. It was developed for Node.js runtime environment in collaboration by facebook, google and others in order to provide consistency, security and improve performance. 
+
+3. What is a React component?  
+
+Your answer: A React component is functionality that is added to a React application. It is sort of like a function in that it provides the structure of the component and then it is invoked from the App.js parent component where additional logic is performed using the child component's structure. 
+
+Researched answer: React components are independent and reusable code that are similar to functions but work independently and return html code. Divides the UI into smaller bits, making it easier to maintain a large codebase.
+
+4. What is the difference between state values and props in React?
+
+Your answer: State values are kept track of by the React application and are updated as required, making the application change dynamically. Props are properties that can be passed from the parent component to the child components in order for them to be able to be referenced between them. Without props, the child component can't access the given value. 
+
+Researched answer: state values are updateable structures containing data whereas the props are static, read-only components that are similar to function arguments.
+
+5. What is the DOM?
+
+Your answer: DOM or document object model is a sort of reversed UI, whereas the UI makes the code readable to humans, the DOm makes it interactable to the computer. An API for the webpage. 
+
+Researched answer: The DOM makes the HTML document more dynamic as it views the document as a set of nodes allowing us to change the nodes by creating events dynamically. 
+
+6. STRETCH: Which is the difference between a div tag and a span tag?
+
+Your answer: A div tag and a span tag are both a way to create a "block" to separate html/js code for organization and styling purposes. The difference between them is strictly semantical in nature.
+
+Researched answer: Div tags are block-level elements and are used to wrap sections of a document for styling. Span tags are inline elements used for styling sections of text or images. 
+
```

Comment:
great job on these answers, Borja! I encourage you to use the research to learn something you didn't already know (i.e. your answer was correct :) ).

Reply:
LOL I did learn, I didn't realize that a span was inline level. I thought they were both block level. I think I didn't convey the learning part very well. Thank you Trish.